lib/tasks/scripts/telemedicine_reports.rb

```diff
@@ -0,0 +1,322 @@
+          "Patients with High BP or Sugar",
+          "Teleconsult Button Clicks",
+          "Teleconsult requests percentage"
+        ]
+
+        facilities_data.each do |state|
+          telemed_clicks = fetch_clicks_for_region(mixpanel_data, state, :state)
+          csv << [
+            state[:state],
+            "",
+            "",
+            state[:count],
+            state[:hwc_and_sc],
+            state[:users],
+            "",
+            state[:telemed_data][:visits],
+            state[:telemed_data][:high_bp],
+            state[:telemed_data][:high_bs],
+            state[:telemed_data][:high_bp_or_bs],
+            telemed_clicks,
+            percentage(telemed_clicks, state[:telemed_data][:high_bp_or_bs])
+          ]
+
+          state[:districts].each do |district|
+            telemed_clicks = fetch_clicks_for_region(mixpanel_data, district, :district)
+            csv << [
+              "",
+              district[:district],
+              "",
+              district[:count],
+              district[:hwc_and_sc],
+              district[:users],
+              "",
+              district[:telemed_data][:visits],
+              district[:telemed_data][:high_bp],
+              district[:telemed_data][:high_bs],
+              district[:telemed_data][:high_bp_or_bs],
+              telemed_clicks,
+              percentage(telemed_clicks, district[:telemed_data][:high_bp_or_bs])
+            ]
+
+            district[:facilities].each do |facility|
+              csv << [
+                "",
+                "",
+                facility[:name],
+                "",
+                "",
+                facility[:users],
+                "",
+                facility[:telemed_data][:visits],
+                facility[:telemed_data][:high_bp],
+                facility[:telemed_data][:high_bs],
+                facility[:telemed_data][:high_bp_or_bs],
+                "",
+                ""
+              ]
+            end
+          end
+        end
+
+        csv << []
+        csv << []
+
+        daily_activity_data = raw_mixpanel_data.group_by { |row| row[:date] }.sort_by { |date, _rows| date }.map { |date, rows|
+          [date.strftime("%d %b %Y"), rows.uniq { |row| row[:user_id] }.count, sum_values(rows, :clicks)]
+        }
+
+        csv << ["Date", "Unique users", "Total TC requests"]
+        daily_activity_data.each do |row|
+          csv << row
+        end
+      end
+    end
+
+    private
+
+    def format_mixpanel_data(mixpanel_data)
+      mixpanel_data.group_by { |row| row[:state] }.map { |state, districts|
+        {state: state,
+         clicks: sum_values(districts, :clicks),
+         districts: districts.group_by { |row| row[:district] }.map { |district, users|
+                      {district: district,
+                       clicks: sum_values(users, :clicks)}
+                    }}
+      }
+    end
+
+    def format_facility_data(facility_data)
```

Comment:
This creates the nested data structure that we iterate over to generate the report